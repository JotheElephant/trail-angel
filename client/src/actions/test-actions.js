import * as actionTypes from './action-types';

const update = (custom, actionType) => {
  return {
    type: actionType,
    custom: custom
  }
}

export const testIncrement = (custom) => {
  // traditional action creator style
  return {
    type: actionTypes.TEST_INCREMENT,
    custom: custom
  }
};

export const testDecrement = (custom) => {
  // traditional action creator style
  return {
    type: actionTypes.TEST_DECREMENT,
    custom: custom
  }
};

export const testIncrementAsync = (custom) => {
  // by returning a function, the Thunk middleware is invoked
  return (dispatch, getState) => {
    // you could do something async here before dispatching
    // or only dispatch conditionally
    setTimeout(() => {
      dispatch({
        type: 'TEST_INCREMENT',
        custom: 'a little late'
      })
    }, 1000);
  };
}