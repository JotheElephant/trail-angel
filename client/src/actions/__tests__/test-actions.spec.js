import * as testActions from '../test-actions';

describe('testActions', () => {
  it('creates a TEST_INCREMENT action', () => {
    expect(testActions.testIncrement()).toMatchSnapshot();
  });

  it('creates a TEST_DECREMENT action', () => {
    expect(testActions.testDecrement()).toMatchSnapshot();
  });

  it('dispatches a TEST_INCREMENT action asynchronously', () => {
    // this will require  mocking of the library (e.g. axios) and probably
    // also redux-mock-store
  });
});