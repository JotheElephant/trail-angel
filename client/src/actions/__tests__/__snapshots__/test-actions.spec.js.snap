exports[`testActions creates a TEST_DECREMENT action 1`] = `
Object {
  "custom": undefined,
  "type": "TEST_DECREMENT",
}
`;

exports[`testActions creates a TEST_INCREMENT action 1`] = `
Object {
  "custom": undefined,
  "type": "TEST_INCREMENT",
}
`;
